src/components/LinkableText.tsx

```diff
@@ -35,23 +36,41 @@ export const LinkableText: React.FC<Props> = ({text, ...props}) => {
   };
 
   // Regular expression to match URLs
-  const urlRegex = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/;
+  const urlRegex = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/c\/[A-Z0-9-]+)?(?:\/.*)?)$/i;
+  const regexPrefix = /(http|https)/;
 
   // Split the text by URLs
   const parts = text.split(' ');
 
+  const checkUrl = useCallback((t: string) => {
+    const [isValid, setValid] = useState(false);
+    if (urlRegex.test(t?.toLocaleLowerCase())) {
+      const urls = `http://${t?.trim().toLocaleLowerCase()}`;
+      checkValidUrl(urls)
+        .then(() => {
+          setValid(true);
+        })
+        .catch(() => {
+          setValid(false);
+        });
+    }
+    return isValid;
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues with this code block:

1. The `useState` hook is being used inside the `useCallback` hook, which is not recommended. Hooks should be called at the top level of your React function. By calling hooks inside functions, loops, conditions, or nested functions, you could potentially violate the rules of hooks.

2. The `checkUrl` function is asynchronous because it calls an asynchronous function `checkValidUrl`. However, it doesn't return a Promise and instead returns a boolean value immediately. This means that the `isValid` state might not be updated in time when it's returned.

3. The `checkUrl` function is called inside the render method, which could lead to unnecessary re-renders and performance issues because it changes state (`setValid`) during rendering.

Here's how you can fix these issues:

```diff
-  const checkUrl = useCallback((t: string) => {
-    const [isValid, setValid] = useState(false);
-    if (urlRegex.test(t?.toLocaleLowerCase())) {
-      const urls = `http://${t?.trim().toLocaleLowerCase()}`;
-      checkValidUrl(urls)
-        .then(() => {
-          setValid(true);
-        })
-        .catch(() => {
-          setValid(false);
-        });
-    }
-    return isValid;
-  }, []);
+  const [isValid, setIsValid] = useState(false);
+
+  const checkUrl = useCallback(async (t: string) => {
+    if (urlRegex.test(t?.toLocaleLowerCase())) {
+      const urls = `http://${t?.trim().toLocaleLowerCase()}`;
+      try {
+        await checkValidUrl(urls);
+        setIsValid(true);
+      } catch {
+        setIsValid(false);
+      }
+    }
+  }, []);
```

You should also move the call to `checkUrl` outside of the render method and use an effect hook to call it when necessary.



<!-- This is an auto-generated comment by OSS CodeRabbit -->